.github/workflows/pull-request.yaml

```diff
@@ -0,0 +1,62 @@
+name: Replicated PR Channel Create
+
+on:
+  pull_request:
+
+permissions: write-all
+
+env:
+  REPLICATED_APP: copia
+
+jobs:
+  push-to-replicated:
+    runs-on: ubuntu-latest
+    outputs:
+      channel-slug: ${{ steps.create-release.outputs.channel-slug }}
+      release-sequence: ${{ steps.create-release.outputs.release-sequence }}
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Define Channel Name
+        run: |
+          FILTERED_REF_NAME=$(echo ${{ github.ref_name }} | sed 's/\//\-/g')
+          echo "CHANNEL_NAME=$FILTERED_REF_NAME" >> $GITHUB_ENV
+          echo "REF_NAME=$FILTERED_REF_NAME-${GITHUB_RUN_ID}${GITHUB_RUN_ATTEMPT}" >> $GITHUB_ENV
+
+      - name: Update the HelmChart kind
+        uses: jacobtomlinson/gha-find-replace@a51bbcd94d000df9ca0fcb54ec8be69aad8374b0 # v3
+        with:
+          include: 'charts/copia/Chart.yaml'
+          find: '$VERSION'
+          replace: ${{ env.REF_NAME }}
+          regex: false
+
+      - name: Package Helm Chart for Replicated
+        id: package-helm-chart
+        working-directory: charts/copia
+        run: |
+          PACKAGE=$(helm package -u . --version=${{ env.REF_NAME }})
+          echo "CHART=$(echo "$PACKAGE" | \
+            grep -oE 'Successfully packaged chart and saved it to: (.+)$' | \
+            sed 's/Successfully packaged chart and saved it to: //')" >> $GITHUB_ENV
+
+      - name: Create Replicated Release
+        id: create-release
+        uses: replicatedhq/compatibility-actions/create-release@5553df9be85049c466907f692e94cfcaf490c926 # v1.5.2
+        with:
+          app-slug: ${{ env.REPLICATED_APP }}
+          api-token: ${{ secrets.REPLICATED_API_KEY }}
+          chart: ${{ env.CHART }}
+          promote-channel: ${{ env.CHANNEL_NAME }}
+          version: ${{ env.REF_NAME }}
+
+      - name: Archive Replicated Channel
+        if: github.event.pull_request.merged == true || 
```

Comment:
1. If no condition follows ||, can it be removed? Or is it supposed to check for some other condition?

2. This workflow will be executed as soon as PR is created with change to appversion in chart.yaml. If we need to update the version in the same PR, seems like this workflow will get triggered multiple times, with REF_NAME being unique for each time this workflow is created. Should we instead trigger pushing into replicate on push event on the main branch? So instead of PRs controlling pushing into replicate, can it the responsibility of the main branch?

Reply:
this should have been moved to the closed.yaml, resolved

Reply:
Do we still need, closed workflow? Replicated is not created in PR, so, when the PR closes, there will be nothing to archive.

Reply:
We should leave it in place. The release will run on both PRs and git tag push, and PR-specific channels should be pruned on close.